Makefile

```diff
@@ -47,7 +47,7 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 		output:rbac:artifacts:config=manifests/v2/base/rbac \
 		output:webhook:artifacts:config=manifests/v2/base/webhook
 
-generate: controller-gen manifests ## Generate apidoc, sdk and code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+generate: go-mod-download controller-gen manifests ## Generate apidoc, sdk and code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
```

Comment:
@astefanutti @kubeflow/wg-training-leads Do we need a separate task for `go mod download` or we can just include this command as part of generate task, for instance:
```
generate: manifests ## Generate apidoc, sdk and code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
        go mod download
	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate/boilerplate.go.txt" paths="./pkg/apis/..."
	hack/update-codegen.sh
	hack/python-sdk/gen-sdk.sh
	hack/python-sdk-v2/gen-sdk.sh
	$(MAKE) apidoc
```

Reply:
I find a separate target slightly more consistent with those for controller-get and Kustomize, but I don't have a strong opinion :)

Reply:
I am fine with both, any thoughts @tenzen-y ?

Reply:
I would vote to @astefanutti solution.